parlai/mturk/core/react_server/dev/components/core_components.jsx

```diff
@@ -822,11 +822,140 @@ class TextResponse extends React.Component {
   }
 }
 
+class FormResponse extends React.Component {
+  // Provide a form-like interface to MTurk interface.
+
+  constructor(props) {
+    super(props);
+    // At this point it should be assumed that task_data
+    // has a field "respond_with_form"
+    let responses = []
+    for (let _ of this.props.task_data["respond_with_form"]){
+      responses.push('');
+    }
+    this.state = {'responses': responses, 'sending': false};
+  }
+
+
+  tryMessageSend() {
+    let form_elements = this.props.task_data["respond_with_form"];
+    let response_data = {};
+    let response_text = "";
+    for (let ind in form_elements){
+      let question = form_elements[ind]["question"];
+      let response = this.state.responses[ind];
+      response_data[question] = response;
+      response_text += question + ": " + response + "\n";
+    }
+
+    if (this.state.textval != '' && this.props.active && !this.state.sending) {
+      this.setState({sending: true});
+      this.props.onMessageSend(
+        response_text, response_data,
+        () => this.setState({'sending': false}));
+    }
+  }
+
+  render() {
+    let form_elements = this.props.task_data["respond_with_form"];
+    const listFormElements= form_elements.map((form_elem, index) => {
+        let question = form_elem["question"];
+        if (form_elem["type"] == "choices"){
+          let choices = [<option></option>].concat(
+            form_elem["choices"].map((option_label) => {
+              return <option>{option_label}</option>
+            }));
+          return (<FormGroup>
```

Comment:
React might be complaining about elements in an array not having unique keys. You should add keys to these components based on some identifier. I think this should be something that the user can tag rather than `question`